extensions/azdata/src/azdata.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import * as path from 'path';
+import * as uuid from 'uuid';
+import * as which from 'which';
+import * as vscode from 'vscode';
+import { HttpClient } from './common/httpClient';
+import * as loc from './localizedConstants';
+import { executeCommand } from './common/childProcess';
+
+/**
+ * Information about an azdata installation
+ */
+export interface IAzdata {
+	path: string,
+	version: string
+}
+
+/**
+ * Finds the existing installation of azdata, or throws an error if it couldn't find it
+ * or encountered an unexpected error.
+ * @param outputChannel Channel used to display diagnostic information
+ */
+export async function findAzdata(outputChannel: vscode.OutputChannel): Promise<IAzdata> {
+	outputChannel.appendLine(loc.searchingForAzdata);
+	try {
+		let azdata: IAzdata | undefined = undefined;
+		switch (process.platform) {
+			case 'darwin':
+				azdata = await findAzdataDarwin(outputChannel);
+				break;
+			case 'win32':
+				azdata = await findAzdataWin32(outputChannel);
+				break;
+			default:
+				azdata = await findSpecificAzdata('azdata', outputChannel);
+		}
+		outputChannel.appendLine(loc.foundExistingAzdata(azdata.path, azdata.version));
+		return azdata;
+	} catch (err) {
+		outputChannel.appendLine(loc.couldNotFindAzdata(err));
+		throw err;
+	}
+}
+
+/**
+ * Downloads the appropriate installer and/or runs the command to install azdata
+ * @param outputChannel Channel used to display diagnostic information
+ */
+export async function downloadAndInstallAzdata(outputChannel: vscode.OutputChannel): Promise<void> {
+	const statusDisposable = vscode.window.setStatusBarMessage(loc.installingAzdata);
+	try {
+		switch (process.platform) {
+			case 'win32': await downloadAndInstallAzdataWin32(outputChannel);
```

Comment:
Should we throw an error in the default case here?

Reply:
Yeah I'm going to do something different here - going to leave this as is for now since I still want to think about the best way of handling these defaults. Will add in a better case here once I add support for the others.